Add conn_track demo for native code generation (#16)
microsoft/ebpf-for-windows-demo: connection_tracker/README.md

> # eBPF For Windows Demo - Connection Tracking

Reviewer: I think we should add a link to this md file in the top level readme also, which lists all the demos in this repo.

1) Native eBPF program generation.
2) The BPF_PROG_TYPE_SOCK_OPS program type.
3) The bpf_printk helper emitting tracing to ETW.
4) The BPF_MAP_TYPE_RINGBUF map type.

The project provides a real-time list of connections that have been completed along with the source, destination, and duration of each connection.

## Demo Steps
### Native eBPF program generation.
Note: These steps are already performed as part of the build process for the project and are repeated here for clarity.

1) Launch a "Developer Command Prompt for VS 2019"
2) For convenience, set an environment variable to point to the root of the eBPF nuget package: ```set EBPF_ROOT=e:\ebpf-for-windows-demo\packages\eBPF-for-Windows.0.1.4\build\native```
3) Change directory to connection_tracker\bpf: ```cd connection_tracker\bpf```
4) Create a directory to hold build output: ```md out```
5) Build the eBPF program to an ELF file: ```clang -g -target bpf -O2 -Werror -I%EBPF_ROOT%\include -c conn_track.c -o out\conn_track.o```
6) Invoke the powershell script to build the native image: ```cd out && powershell -NonInteractive -ExecutionPolicy Unrestricted %EBPF_ROOT%\bin\Convert-BpfToNative.ps1 -ProgramName conn_track.o```

> 7) This produces the native eBPF program conn_track.sys.

Reviewer 2: nit: this line ends with a period and the others don't.  In contrast, lines 5-8 all end in periods as do 26-31, etc.

Section headings on the other hand I think should not end in periods (25, 33, 13 etc should remove it)

Author: I think I fixed all cases. All sentences should end with periods.

8) Netsh tools can be used to examine the native eBPF program: ```netsh ebpf show sections conn_track.sys```

### BPF_PROG_TYPE_SOCK_OPS and BPF_MAP_TYPE_RINGBUF demo.
1) Build the ebpf-for-windows-demo as outlined in [Getting Started](https://github.com/microsoft/ebpf-for-windows-demo/blob/main/docs/GettingStarted.md).
2) Install eBPF-For-Windows on the target machine.
3) Copy conn_track.sys and conn_tracker.exe to the target machine.
4) Launch conn_tracker.exe.
5) Launch a browser and navigate to any website.
6) Connection tracker will then show the list of connections.

### Viewing bpf_printk output.

> 1) Start a ETW session and add the eBPF-For-Windows provider: ```tracelog -start MyTrace -guid ebpf-printk.guid -rt```

Reviewer 2: ```suggestion
1) Start an ETW session and add the eBPF-For-Windows provider: ```tracelog -start MyTrace -guid ebpf-printk.guid -rt```
```
